packages/core/src/shareback/create-and-upload-extrinsic-object.ts

```diff
@@ -0,0 +1,95 @@
+import { createExtrinsicObjectXml } from "../external/carequality/dq/create-metadata-xml";
+import { createPatientUniqueId } from "../external/carequality/shared";
+import { S3Utils } from "../external/aws/s3";
+import { DocumentReference, Organization } from "@medplum/fhirtypes";
+import { out } from "../util/log";
+
+const { log } = out("Core Create and Upload Extrinsic Object");
+
+export async function createAndUploadDocumentdMetadataFile({
+  s3Utils,
+  cxId,
+  patientId,
+  docId,
+  size,
+  docRef,
+  metadataFileName,
+  destinationBucket,
+  mimeType,
+}: {
+  s3Utils: S3Utils;
+  cxId: string;
+  patientId: string;
+  docId: string;
+  hash: string;
+  size: string;
+  docRef: DocumentReference;
+  metadataFileName: string;
+  destinationBucket: string;
+  mimeType: string;
+}): Promise<void> {
+  const createdTime = new Date().toISOString();
+  const uniquePatientId = createPatientUniqueId(cxId, patientId);
+
+  const title = docRef?.description;
+  const classCode = docRef?.type;
+  const practiceSettingCode = docRef?.context?.practiceSetting;
+  const healthcareFacilityTypeCode = docRef?.context?.facilityType;
+  const organization: Organization | undefined = docRef?.contained?.find(
+    (resource): resource is Organization => resource.resourceType === "Organization"
+  );
+  log(`Creating metadata file for docId: ${docId}`);
+  const extrinsicObjectXml = createExtrinsicObjectXml({
+    createdTime,
+    size,
+    patientId: uniquePatientId,
+    organization,
+    classCode,
+    practiceSettingCode,
+    healthcareFacilityTypeCode,
+    documentUniqueId: docId,
+    title,
+    mimeType,
+  });
+
+  log(`Uploading metadata to S3 with key: ${metadataFileName}`);
+  await s3Utils.uploadFile(destinationBucket, metadataFileName, Buffer.from(extrinsicObjectXml));
+}
+
+export async function createAndUploadCDAMetadataFile({
+  s3Utils,
+  cxId,
+  patientId,
+  docId,
+  size,
+  organization,
+  metadataFileName,
+  destinationBucket,
+  mimeType,
+}: {
+  s3Utils: S3Utils;
+  cxId: string;
+  patientId: string;
+  docId: string;
+  hash: string;
+  size: string;
+  organization: Organization;
+  metadataFileName: string;
+  destinationBucket: string;
+  mimeType: string;
+}): Promise<void> {
+  const createdTime = new Date().toISOString();
+  const uniquePatientId = createPatientUniqueId(cxId, patientId);
+  log(`Creating metadata file for docId: ${docId}`);
+  const extrinsicObjectXml = createExtrinsicObjectXml({
+    createdTime,
+    size,
+    patientId: uniquePatientId,
+    documentUniqueId: docId,
+    organization,
+    mimeType,
+  });
```

Comment:
We've been talking about the issues with storing pre-processed metadata on S3, why are we creating code with the same approach?

Can we store a JSON with most of the information about the doc ref and have the code that uses it to generate the XML on-the-fly (then we can change the code w/o worrying about previously generated/stored data and migrations)?

Reply:
Perhaps, I'm missing some context here. Would appreciate a sync regarding this or just a clearer direction on what changes are expected

Reply:
The issue is we need to change the inbound document query logic. We can't just change things here. 

https://github.com/metriport/metriport-internal/issues/1674
thread with [context](https://metriport.slack.com/archives/C065BLRBUDQ/p1712344859986849) and if we were to change it,  would go to querying FHIR server for doc Refs, not storing JSON on s3 given low patient volume

Reply:
I removed the duplicated function